api.planx.uk/server.js

```diff
@@ -490,18 +500,63 @@ app.post("/download-application", async (req, res, next) => {
   }
 });
 
-app.post("/sign-s3-upload", async (req, res, next) => {
-  if (!req.body.filename) next({ status: 422, message: "missing filename" });
+app.post("/private-file-upload", multer().single('file'), async (req, res, next) => {
+  if (!req.body.filename) return next({ status: 422, message: "missing filename" });
+  if (!req.file) return next({ status: 422, message: "missing file" });
 
   try {
-    const { fileType, url, acl } = await signS3Upload(req.body.filename);
+    const fileResponse = await uploadPrivateFile(req.file, req.body.filename);
 
-    res.json({
-      upload_to: url,
-      public_readonly_url_will_be: url.split("?")[0],
-      file_type: fileType,
-      acl,
-    });
+    res.json(fileResponse);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post("/public-file-upload", multer().single('file'), async (req, res, next) => {
+  if (!req.body.filename) return next({ status: 422, message: "missing filename" });
+  if (!req.file) return next({ status: 422, message: "missing file" });
+
+  try {
+    const fileResponse = await uploadPublicFile(req.file, req.body.filename);
+
+    res.json(fileResponse);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get("/file/public/:fileKey/:fileName", async (req, res, next) => {
+  const filePath = buildFilePath(req.params.fileKey, req.params.fileName);
+
+  if (!filePath) {
+    return next({ status: 404, message: "file not found" });
+  }
+
+  try {
+    const { body, headers, isPrivate } = await getFileFromS3(filePath);
+
+    if (isPrivate) return next({ status: 400, message: "bad request" });
+
+    res.set(headers);
+    res.send(body);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get("/file/private/:fileKey/:fileName", useFilePermission, async (req, res, next) => {
+  const filePath = buildFilePath(req.params.fileKey, req.params.fileName);
+
+  if (!filePath) {
+    return next({ status: 404, message: "file not found" });
+  }
+
+  try {
+    const { body, headers } = await getFileFromS3(filePath)
+
+    res.set(headers);
+    res.send(body);
```

Comment:
nit: I would personally love to see only endpoints in `server.js` and then add these methods/logic to existing `s3.js` file or a new file. same goes for test cases you've added to `server.test.js`. API does all the things and I've been findng sub-files helpful for mental compartmentalization & quicker investigation/troubleshooting